src/third_party/wiredtiger/src/conn/conn_dhandle.c

```diff
@@ -548,6 +557,12 @@ __wt_conn_btree_apply(WT_SESSION_IMPL *session, const char *uri,
 
 	conn = S2C(session);
 
+	/* If optimized sweeping is enabled then sweep session dhandles
+	 * once only instead of once for each dhandle when going through each of them
+	 */
+	if (F_ISSET(S2C(session), WT_CONN_BT_APPLY_ESWEEP | WT_CONN_CKPT_PREP_NOSWEEP))
```

Comment:
I am curious why there are two flags. When would you want one to be on and not the other?

Reply:
Are you saying that we should use just one flag but have 2 configs and set the flag according to the 2 configs ?

I was trying to separate them because the apply_all is used by a lot more things so if it causes problem, we can disable it and it will avoid sweeps only in the checkpoint code which gives some benefit but not full benefit.

Reply:
I meant that you would have one user-settable config that sets one flag. That flag would then be used in the places where we want to disable sweep.

If we run into problems, it seems like we would just turn it all off. Not a big deal either way, though.